src/main/scala/org/ergoplatform/nodeView/wallet/scanning/ScanningPredicateJsonCodecs.scala

```diff
@@ -0,0 +1,55 @@
+package org.ergoplatform.nodeView.wallet.scanning
+
+import io.circe.Json
+import org.ergoplatform.ErgoBox
+import org.ergoplatform.ErgoBox.RegisterId
+import scorex.util.encode.Base16
+import io.circe.{Decoder, Encoder}
+import io.circe.generic.auto._
+import io.circe.syntax._
+import org.ergoplatform.http.api.ApiCodecs
+
+
+object ScanningPredicateJsonCodecs extends ApiCodecs {
+
+  implicit val scanningPredicateEncoder: Encoder[ScanningPredicate] = {
+    case cp: ContainsScanningPredicate =>
+      Json.obj("predicate" -> "contains".asJson, "register" -> cp.regId.asJson, "bytes" -> Base16.encode(cp.bytes).asJson)
+    case ep: EqualsScanningPredicate =>
+      Json.obj("predicate" -> "equals".asJson, "register" -> ep.regId.asJson, "bytes" -> Base16.encode(ep.bytes).asJson)
+    case cap: ContainsAssetPredicate =>
+      Json.obj("predicate" -> "containsAsset".asJson, "assetId" -> Base16.encode(cap.assetId).asJson)
+    case and: AndScanningPredicate =>
+      Json.obj("predicate" -> "and".asJson, "args" -> and.subPredicates.asJson)
+    case or: OrScanningPredicate =>
+      Json.obj("predicate" -> "or".asJson, "args" -> or.subPredicates.asJson)
+  }
+
+  implicit val scanningPredicateDecoder: Decoder[ScanningPredicate] = { implicit cursor =>
+    cursor.downField("predicate").as[String].flatMap {
+      case predicate@(_: String) if predicate == "containsAsset" =>
```

Comment:
```suggestion
      case  "containsAsset" =>
```
This should also work.